R/read_dataset_json.R

```diff
@@ -3,17 +3,31 @@
 #' This function validated a dataset JSON file on disk against the Dataset JSON schema, and if valid
 #' returns a datasetjson object
 #'
-#' @param file File path on disk
+#' @param file File path on disk, or a pre-loaded Dataset JSON file in a single element character string
 #'
 #' @return datasetjson object
 #' @export
 #'
 #' @examples
+#' # Read from disk
+#' \dontrun{
+#'   dat <- read_dataset_json("path/to/file.json")
+#' }
 #'
-#' # TODO:
+#' # Read from an already imported character vector
+#' ds_json <- dataset_json(iris, "IG.IRIS", "IRIS", "Iris", iris_items)
+#' js <- write_dataset_json(ds_json)
+#' dat <- read_dataset_json(js)
 read_dataset_json <- function(file) {
   # Validate the input file against the schema
-  jsonvalidate::json_validate(file, schema_1_0_0, engine="ajv")
+  valid <- jsonvalidate::json_validate(file, schema_1_0_0, engine="ajv")
+
+  if (!valid) {
+    stop(paste(
+      "Dataset JSON file is invalid per the JSON schema.",
+      "Use datasetjson::validate_dataset_json() to see details"),
+      call.=FALSE)
+  }
```

Comment:
Do we want to give them the code to run?
```suggestion
    if (!valid) {
    stop(paste0(
      "Dataset JSON file is invalid per the JSON schema. ",
      "Run datasetjson::validate_dataset_json(",substitute(file),") to see details"),
      call.=FALSE)
  }
```

Reply:
Ok I updated this per your suggestion - I just had to edit it a bit to get the print right. I also added an error check for the writer because we didn't have one.